homeassistant/components/fmi/weather.py

```diff
@@ -0,0 +1,183 @@
+"""Support for retrieving meteorological data from FMI (Finnish Meteorological Institute)."""
+from dateutil import tz
+
+from homeassistant.components.weather import (
+    ATTR_FORECAST_CONDITION,
+    ATTR_FORECAST_PRECIPITATION,
+    ATTR_FORECAST_TEMP,
+    ATTR_FORECAST_TIME,
+    ATTR_FORECAST_WIND_BEARING,
+    ATTR_FORECAST_WIND_SPEED,
+    ATTR_WEATHER_HUMIDITY,
+    ATTR_WEATHER_PRESSURE,
+    WeatherEntity,
+)
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import get_weather_symbol
+from .const import _LOGGER, ATTRIBUTION, COORDINATOR, DOMAIN, MANUFACTURER, NAME
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add an FMI weather entity from a config_entry."""
+    name = config_entry.data[CONF_NAME]
+
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
+
+    async_add_entities([FMIWeatherEntity(name, coordinator)], False)
+
+
+class FMIWeatherEntity(CoordinatorEntity, WeatherEntity):
+    """Define an FMI Weather Entity."""
+
+    def __init__(self, name, coordinator):
+        """Initialize FMI weather object."""
+        super().__init__(coordinator)
+        self._name = name
+        self._attrs = {}
+        self._unit_system = "Metric"
+        self._fmi = coordinator
```

Comment:
You don't need this, as the `CoordinatorEntity` already provides `self.coordinator`.